In [5]:
import os
import pandas as pd
print(os.getcwd())

c:\Users\lberm\OneDrive\Documents\Github\gestalt_app\src\data


## Loading and Data Inspection

In [41]:
data_path = r"Question_Embedding_20241230.csv"
df =pd.read_csv(data_path)
print(df.columns)
df.head()

Index(['Unnamed: 0.1', 'Question Title', 'Unnamed: 0', 'question.html',
       'server.js', 'solution.html', 'server.py', 'properties.js',
       'info1.json', 'server_trap.js', 'server1.py', 'server2.py', 'test1.py',
       'server3.py', '.DS_Store', 'question', 'question_embedding', 'uuid',
       'title', 'stem', 'topic', 'tags', 'prereqs', 'isAdaptive', 'createdBy',
       'qType', 'nSteps', 'updatedBy', 'difficulty', 'codelang', 'resources',
       'stepType', 'dificulty', 'embeddings-3-small', 'info.json', '_merge'],
      dtype='object')


,Unnamed: 0.1,Question Title,Unnamed: 0,question.html,server.js,solution.html,server.py,properties.js,info1.json,server_trap.js,...,nSteps,updatedBy,difficulty,codelang,resources,stepType,dificulty,embeddings-3-small,info.json,_merge
0,4,1DHeatConduction,202.0,\r\n<pl-question-panel>\r\n <pl-figure file...,const math = require('mathjs');\r\n// const ma...,NaN,NaN,NaN,NaN,NaN,...,1.0,NaN,1.0,javascript,NaN,NaN,NaN,"[0.015802497044205666, -0.03678983449935913, 0...",NaN,left_only
1,5,1DHeatConduction2,203.0,\r\n<pl-question-panel>\r\n <pl-figure file...,const math = require('mathjs');\r\n// const ma...,NaN,NaN,NaN,NaN,NaN,...,1.0,NaN,1.0,javascript,NaN,NaN,NaN,"[0.0229609664529562, -0.07461682707071304, -0....",NaN,left_only
2,201,3DMoment1,30.0,<pl-question-panel>\r\n <pl-figure file-name=...,NaN,<pl-solution-panel>\r\n <pl-figure file-nam...,NaN,NaN,NaN,NaN,...,1.0,NaN,1.0,none,NaN,NaN,NaN,"[-0.02334902435541153, -0.0003407486656215042,...",NaN,left_only
3,202,3DMoment2,31.0,<pl-question-panel>\r\n <pl-figure file-name=...,NaN,<pl-solution-panel>\r\n <pl-figure file-nam...,NaN,NaN,NaN,NaN,...,1.0,NaN,1.0,none,NaN,NaN,NaN,"[0.017233634367585182, -0.022736169397830963, ...",NaN,left_only
4,200,3DMoment3,32.0,<pl-question-panel>\r\n <pl-figure file-name=...,NaN,<pl-solution-panel>\r\n <pl-figure file-nam...,NaN,NaN,NaN,NaN,...,1.0,NaN,1.0,none,NaN,NaN,NaN,"[0.043644580990076065, -0.013504264876246452, ...",NaN,left_only


## Clean up 

In [42]:
columns_to_keep = ["Question Title","question.html","server.js","solution.html","server.py","properties.js","info.json","isAdaptive","embeddings-3-small"]
df = df[columns_to_keep]
df.head()

,Question Title,question.html,server.js,solution.html,server.py,properties.js,info.json,isAdaptive,embeddings-3-small
0,1DHeatConduction,\r\n<pl-question-panel>\r\n <pl-figure file...,const math = require('mathjs');\r\n// const ma...,NaN,NaN,NaN,NaN,True,"[0.015802497044205666, -0.03678983449935913, 0..."
1,1DHeatConduction2,\r\n<pl-question-panel>\r\n <pl-figure file...,const math = require('mathjs');\r\n// const ma...,NaN,NaN,NaN,NaN,True,"[0.0229609664529562, -0.07461682707071304, -0...."
2,3DMoment1,<pl-question-panel>\r\n <pl-figure file-name=...,NaN,<pl-solution-panel>\r\n <pl-figure file-nam...,NaN,NaN,NaN,False,"[-0.02334902435541153, -0.0003407486656215042,..."
3,3DMoment2,<pl-question-panel>\r\n <pl-figure file-name=...,NaN,<pl-solution-panel>\r\n <pl-figure file-nam...,NaN,NaN,NaN,False,"[0.017233634367585182, -0.022736169397830963, ..."
4,3DMoment3,<pl-question-panel>\r\n <pl-figure file-name=...,NaN,<pl-solution-panel>\r\n <pl-figure file-nam...,NaN,NaN,NaN,False,"[0.043644580990076065, -0.013504264876246452, ..."


In [85]:
row_num,col_num = df.shape
adaptive_questions_count = df["isAdaptive"].sum()
print(f"""The number of adaptive questions is {adaptive_questions_count} \n while non adaptive questions are {row_num-adaptive_questions_count}
"""
)
py_filtered = df[(df["isAdaptive"]) & (df["server.py"].notna())]
js_filtered = df[(df["isAdaptive"]) & (df["server.js"].notna())]



The number of adaptive questions is 158 
 while non adaptive questions are 162



'\r\n<pl-question-panel>\r\n    <pl-figure file-name="1DHeatConduction.png"></pl-figure>\r\n    <p> Determine the heat rate passing through the wall of an industrial furnace constructed with a thickness\r\n    of $ {{params.width}} {{params.unitsDist}} $ and a thermal\r\n    conductivity of $\\ {{params.k}} {{params.unitsThermalConductivity}}$.\r\n    During steady state operations the inner and outter temperatures were measured to be $\\ {{params.T1}} {{params.unitsTemperature}} $ and $\\ {{params.T2}} {{params.unitsTemperature}} $ repectively.\r\n     The height of the furnace is measured to be$\\ {{params.height}} {{params.unitsDist}} $ and the length of the furnace is $\\ {{params.length}}  {{params.unitsDist}}$ </p>\r\n</pl-question-panel>\r\n\r\n<pl-number-input answers-name="heatFlux" comparison="sigfig" digits="3" label="The Heat Flux= "></pl-number-input>\r\n<pl-number-input answers-name="heatRate" comparison="sigfig" digits="3" label="The Heat Rate = "></pl-number-input>\r\n'

## Available Tags
The following are the available tags used in most questions

In [115]:
import re
pattern = r'</pl-[a-z\-]+>'
unique_tags = set()
filtered = df[~df["isAdaptive"].isna()]
for text in filtered["question.html"]:
    if isinstance(text, str):
        result = re.findall(pattern, text)
        unique_tags.update(result)
print(unique_tags)

{'</pl-prompt>', '</pl-answer>', '</pl-question-panel>', '</pl-matrix-input>', '</pl-mca-adaptive>', '</pl-static-text>', '</pl-matrix-latex>', '</pl-multiple-choice>', '</pl-mcq-adaptive>', '</pl-quest>', '</pl-checkbox>', '</pl-number-input-fixed>', '</pl-figure>', '</pl-number-input>'}


## Notes
- info.json: Originally held question metadata, now empty. Not essential for generation but should be repopulated.
- question.html: HTML template that renders the student-facing question (figures, inputs, choices, etc.).
- server files: Backend logic written in either JavaScript or Python.
-Total questions: 320 (used for semantic search); not guaranteed to reflect the current total.
- Adaptive questions: 158 — dynamically generated (e.g. numerical computations).
- Non-adaptive questions: 162 — static (e.g. multiple choice).
- Code files: 7 Python files, 139 JavaScript files related to questions.